### DISCUSSION TITLE
Is it psossible to make version selection a dropdown instead of text entry in Swagger?

### DISCUSSION THREAD

**Discussion by voroninp** [dotnet/aspnet-api-versioning#1088] [Q&A] [answered]
![image](https://github.com/dotnet/aspnet-api-versioning/assets/9202028/0ab835f4-3999-4b16-9ffc-149871418732)

I am curious whether it's possible to make it a fixed list of avialable versions.

**Comment by voroninp** [comment]
Or maybe it can have one fixed value defined by the selection:
![image](https://github.com/dotnet/aspnet-api-versioning/assets/9202028/90a60df7-e0f1-4602-a5f2-ca7394ec5d18)

**Comment by commonsensesoftware** [comment, marked as answer]
Yes - it is possible. In the example project, change the line:

https://github.com/dotnet/aspnet-api-versioning/blob/3fc071913dcded23eeb5ebe55bca44f3828488bf/examples/AspNetCore/WebApi/OpenApiExample/SwaggerDefaultValues.cs#L59

to

```c#
parameter.Schema.Enum = [OpenApiAnyFactory.CreateFromJson( json )];
```

This will yield the API version as an enumeration.

![image](https://github.com/dotnet/aspnet-api-versioning/assets/11765008/7d8c5ab2-b622-4946-852f-ae87bb0c03db)

It should be noted that the documented API may **not** be representative of all API versions. The model, parameters, and so on may be different across API versions, even if the URL is the same (which it _should_ be). You can use the _supported_ API versions to produce a full list, but realize that this is for the **entire** API (ex: the _Orders_ API) and not a specific endpoint. It is possible that an endpoint may not exist in a supported API version.

If that doesn't apply to you or you don't care about that limitation, you can list all _supported_ API versions like so:

```c#
var metadata = apiDescription.ActionDescriptor.GetApiVersionMetadata();
var model = metadata.Map( ApiVersionMapping.Explicit | ApiVersionMapping.Implicit );

parameter.Schema.Enum =
    model.SupportedApiVersions.Select(
        version => OpenApiAnyFactory.CreateFromJson(
                    JsonSerializer.Serialize(
                        version.ToString(),
                        modelMetadata.ModelType ) ) ).ToImmutableArray();

parameter.Schema.Default =
    OpenApiAnyFactory.CreateFromJson(
        JsonSerializer.Serialize(
            description.DefaultValue,
            modelMetadata.ModelType ) );
```

This will also continue to set the default value so that the default selection is whichever API version you are currently viewing.

![image](https://github.com/dotnet/aspnet-api-versioning/assets/11765008/a9a9411e-9a76-4b89-892f-22cb9a811773)

I hope that helps.
